### PR TITLE
#11 Validations on SignUp Screen

### PR DESCRIPTION
**Pull request by OrlandoAlPe**
Completed the following validations for the SignUp Screen:

- Name TextInput
- Email TextInput
- Password TextInpt
- Term and policy agreement checkbox

The app verifies if those validations are passed or not with an alert message.

Screenshot_1618513521

Screenshot_1618513683

Firebase SignUp services are Still pending.

### CODE REVIEW

**Review by jkzasori — COMMENTED** [review]
Are you using codeclima? I'm asking because I can't see at the bottom...

**Comment by Coreandrum1** [comment]
> Are you using codeclima? I'm asking because I can't see at the bottom...

**Comment by OrlandoAlPe** [comment]
> Are you using codeclima? I'm asking because I can't see at the bottom...
The repostory is private so we can´t implement code climate yet.

**Review by francisco-sle — COMMENTED** [review]
LGTM

File: reservacionVuelo/Screens/SignUp.js [hunk 1]
```diff
@@ -1,59 +1,138 @@
...
-import CheckBox from '@react-native-community/checkbox';
-import Icon from 'react-native-vector-icons/EvilIcons';
+import {Alert, Text, TouchableHighlight, View} from 'react-native';
+import Icon from 'react-native-vector-icons/FontAwesome';
 import styles from '../components/styles';
+import {CustomCheckbox} from '../components/checkbox';
+import {CustomTextInput} from '../components/TextInput';
+
+const emailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+const nameRegex = /^[A-Za-z]+$/;
```
Comment (Joaquin-Niembro) [inline reservacionVuelo/Screens/SignUp.js]: Put this expressions in a constants file

File: reservacionVuelo/Screens/SignUp.js [hunk 2]
```diff
@@ -1,59 +1,138 @@
...
+        <CustomTextInput
+          textInputTitle={'First Name'}
+          onChangeText={text =>
+            this.setState({
+              name: text,
+              nameStatus: nameRegex.test(this.state.name),
+            })
+          }
+          checkError={this.state.nameStatus === false}
+          errorMessage={'Write a valid name please'}
```
Comment (Joaquin-Niembro) [inline reservacionVuelo/Screens/SignUp.js]: Put the errorMessage in a constants file

File: reservacionVuelo/Screens/SignUp.js [hunk 3]
```diff
@@ -1,59 +1,138 @@
...
-        </View>
+        <CustomCheckbox
+          value={this.state.checkBox}
+          onValueChange={() => this.setState({checkBox: !this.state.checkBox})}
+          text={'I agree to the '}
+          labelText={'Terms'}
+          secondaryText={' and '}
+          secondaryLabelText={'Privacy Policy.'}
+          check={this.state.checkBox}
+          errorMessage={'Terms and privacy policy must be checked'}
```
Comment (Joaquin-Niembro) [inline reservacionVuelo/Screens/SignUp.js]: Put these messages and texts in a constants file

File: reservacionVuelo/components/checkbox.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import CheckBox from '@react-native-community/checkbox';
+import React, {Component} from 'react';
+import {Text, View} from 'react-native';
+import styles from './styles';
+
+export class CustomCheckbox extends Component {
+  constructor() {
+    super();
+  }
```
Comment (Joaquin-Niembro) [inline reservacionVuelo/components/checkbox.js]: Remove the constructor if its not in use

**Review by Joaquin-Niembro — CHANGES_REQUESTED** [review]

**Review by Joaquin-Niembro — APPROVED** [review]

**Review by francisco-sle — APPROVED** [review]

**Comment by DiegoSpielzeug** [comment]
why don't you have CodeClimate and Deepscan?

**Review by jkzasori — APPROVED** [review]

**Pull request merged**